pkg/apis/policy/v1alpha1/upstreamtrafficsetting.go

```diff
@@ -193,6 +200,227 @@ type HTTPLocalRateLimitSpec struct {
+}
+
+// GenericKeyDescriptorEntry defines a descriptor entry with a static
+// key-value pair.
+type GenericKeyDescriptorEntry struct {
+	// Value defines the descriptor entry's value.
+	Value string `json:"value"`
+
+	// Key defines the descriptor entry's key.
+	// Defaults to 'generic_key'.
+	// +optional
+	Key string `json:"key,omitempty"`
+}
+
+// RemoteAddressDescriptorEntry defines a descriptor entry with
+// key 'remote_address' and value equal to the client's IP address
+// derived from the x-forwarded-for header.
+type RemoteAddressDescriptorEntry struct{}
+
+// RequestHeaderDescriptorEntry defines a descriptor entry that is generated only
+// when the request header matches the given header name. The value of the descriptor
+// entry is derived from the value of the header present in the request.
+type RequestHeaderDescriptorEntry struct {
+	// Name defines the name of the header used to look up the descriptor entry's value.
+	Name string `json:"name"`
+
+	// Key defines the descriptor entry's key.
+	Key string `json:"key"`
+}
+
+// HeaderValueMatchDescriptorEntry defines the descriptor entry that is generated
+// when the request header matches the given HTTP header match criteria.
+type HeaderValueMatchDescriptorEntry struct {
+	// Value defines the descriptor entry's value.
+	Value string `json:"value"`
+
+	// Headers defines the list of HTTP header match criteria.
+	Headers []HTTPHeaderMatcher `json:"headers"`
+
+	// Key defines the descriptor entry's key.
+	// Defaults to 'header_match'.
+	// +optional
+	Key string `json:"key,omitempty"`
+
+	// ExpectMatch defines whether the request must match the given
+	// match criteria for the descriptor entry to be generated.
+	// If set to false, a descriptor entry will be generated when the
+	// request does not match the match criteria.
+	// Defaults to true.
+	// +optional
+	ExpectMatch *bool `json:"expectMatch,omitempty"`
+}
+
+// HTTPHeaderMatcher defines the HTTP header match criteria.
+// Only one of Exact, Prefix, Suffix, Regex, Contains, Present may be set.
+type HTTPHeaderMatcher struct {
+	// Name defines the name of the header to match.
+	Name string `json:"name"`
+
+	// Exact defines the exact value to match.
+	// +optional
+	Exact string `json:"exact,omitempty"`
+
+	// Prefix defines the prefix value to match.
+	// +optional
+	Prefix string `json:"prefix,omitempty"`
+
+	// Suffix defines the suffix value to match.
+	// +optional
+	Suffix string `json:"suffix,omitempty"`
+
+	// Regex defines the regex value to match.
+	// +optional
+	Regex string `json:"regex,omitempty"`
+
+	// Contains defines the substring value to match.
+	// +optional
+	Contains string `json:"contains,omitempty"`
+
+	// Present defines whether the request matches the criteria
+	// when the header is present. If set to false, header match
+	// will be performed based on whether the header is absent.
+	Present *bool `json:"present,omitempty"`
```

Comment:
Is there a default?

Reply:
No. One of these must be set explicitly to either true/false.